gossip/node.go

```diff
@@ -336,6 +337,24 @@ func (n *Node) publishCompletedRound(ctx context.Context) error {
 	return n.pubsub.Publish(n.notaryGroup.ID, conf.Data())
 }
 
+func (n *Node) storeAbrBlocks(ctx context.Context, abr *services.AddBlockRequest) error {
+	sw := safewrap.SafeWrap{}
+
+	for _, nodeBytes := range abr.State {
+		stateNode := sw.Decode(nodeBytes)
+		if sw.Err != nil {
+			return fmt.Errorf("error decoding abr state: %v", sw.Err)
+		}
+
+		err := n.dagStore.Add(ctx, stateNode)
```

Comment:
In addition to @tobowers suggestion about processing this in the bg, I believe `dagStore` also has an `AddMany` which will batch writes given the underlying datastore supports it